Mathlib/SetTheory/Cardinal/Basic.lean

```diff
@@ -454,20 +468,27 @@ theorem mk_prod (α : Type u) (β : Type v) : (#α × β) = lift.{v, u} (#α) *
 #align cardinal.mk_prod Cardinal.mk_prod
 
 private theorem mul_comm' (a b : Cardinal.{u}) : a * b = b * a :=
-  induction_on₂ a b fun α β => mk_congr <| Equiv.prodComm α β
-#align cardinal.mul_comm' cardinal.mul_comm'
+  inductionOn₂ a b fun α β => mk_congr <| Equiv.prodComm α β
+-- #align cardinal.mul_comm' Cardinal.mul_comm'
 
 /-- The cardinal exponential. `#α ^ #β` is the cardinal of `β → α`. -/
-instance : Pow Cardinal.{u} Cardinal.{u} :=
-  ⟨map₂ (fun α β => β → α) fun α β γ δ e₁ e₂ => e₂.arrowCongr e₁⟩
+instance instPowCardinal : Pow Cardinal.{u} Cardinal.{u} :=
+  ⟨map₂ (fun α β => β → α) fun _ _ _ _ e₁ e₂ => e₂.arrowCongr e₁⟩
 
+-- -- Porting note: TODO remove this and deal with the error message concerning `infixr` and `@`.
+-- set_option quotPrecheck false
+
+-- Porting note: I think this is not needed anymore.
```

Comment:
Why do you think so?

Reply:
Tbh I don't fully understand how this "workaround" notation worked in lean3, but it seems that it breaks everything in lean4 and I'm not sure the problem it "worked around" in lean2 still persists in lean4 . if so then maybe there needs to be a different workaround.

For now I adapted the porting note reflecting this and changed to notation to `^'` to avoid the clash with `^`. If you have a deeper insight, feel free to adapt it